### ISSUE TITLE
Lengths and other relations in data

### ISSUE THREAD

**Issue by KOLANICH** [kaitai-io/kaitai_struct#29]
Is it possible to define a structure where the first field is the count (let's call it ```n```) followed by array of ```n``` items of some smaller structure followed by the rest of  structure.
In pseudo-c++
```cpp
struct a{
//.....
};
struct b{
size_t count;
a data[count];
//rest of structure
}
```
Also variants are needed, where the type of field depends on combinational function of bit flags. For example if we process network packets, their content depends on the header. This can be japped to the language using polymophism.

**Comment by GreyCat** [comment]
Yeah, it's possible right now using [repeat](https://github.com/kaitai-io/kaitai_struct/wiki/Attribute-description#repeat). The rough translation of your example in .ksy would be:

```yaml
types:
  a:
    # .....
  b:
    seq:
      - id: count
        type: u4 # or whatever size_t maps to
      - id: data
        type: a
        repeat: expr
        repeat-expr: count
```

Conditionals are also supported. [if](https://github.com/kaitai-io/kaitai_struct/wiki/Attribute-description#if) is available in v0.5 (stable release, as of now), newer versions would include switch to avoid writing lots of `if`s.

**Comment by KOLANICH** [comment]
OK, thank you.
